include/highfive/bits/H5Converter_misc.hpp

```diff
@@ -1,419 +1,750 @@
-struct data_converter<boost::numeric::ublas::matrix<T>, void>
-    : public container_converter<boost::numeric::ublas::matrix<T>> {
-    using Matrix = boost::numeric::ublas::matrix<T>;
-    using value_type = typename inspector<T>::base_type;
-
-    inline data_converter(const DataSpace& space)
-        : container_converter<Matrix>(space) {
-        assert(space.getDimensions().size() == 2);
-    }
-
-    inline value_type* transform_read(Matrix& array) {
-        boost::array<std::size_t, 2> sizes = {{array.size1(), array.size2()}};
-        auto&& _dims = this->_space.getDimensions();
-        if (std::equal(_dims.begin(), _dims.end(), sizes.begin()) == false) {
-            array.resize(_dims[0], _dims[1], false);
-            array(0, 0) = 0;  // force initialization
+    static size_t getSize(const std::vector<size_t>& dims) {
+        return compute_total_size(dims);
+    }
+
+    static void prepare(type& /* val */, const std::vector<size_t>& dims) {
+        if (dims[0] > N) {
+            std::ostringstream os;
+            os << "Size of std::array (" << N << ") is too small for dims (" << dims[0] << ").";
+            throw DataSpaceException(os.str());
         }
-        return &(array(0, 0));
     }
 
-    inline const value_type* transform_write(const Matrix& array) const noexcept {
-        return &(array(0, 0));
+    static hdf5_type* data(type& val) {
+        return inspector<value_type>::data(val[0]);
+    }
+
+    static const hdf5_type* data(const type& val) {
+        return inspector<value_type>::data(val[0]);
+    }
+
+    static void serialize(const type& val, hdf5_type* m) {
+        size_t subsize = compute_total_size(inspector<value_type>::getDimensions(val[0]));
+        for (auto& e: val) {
+            inspector<value_type>::serialize(e, m);
+            m += subsize;
+        }
+    }
+
+    static void unserialize(const hdf5_type* vec_align,
+                            const std::vector<size_t>& dims,
+                            type& val) {
+        if (dims[0] != N) {
+            std::ostringstream os;
+            os << "Impossible to pair DataSet with " << dims[0] << " elements into an array with "
+               << N << " elements.";
+            throw DataSpaceException(os.str());
+        }
+        std::vector<size_t> next_dims(dims.begin() + 1, dims.end());
+        size_t next_size = compute_total_size(next_dims);
+        for (size_t i = 0; i < dims[0]; ++i) {
+            inspector<value_type>::unserialize(vec_align + i * next_size, next_dims, val[i]);
+        }
     }
 };
-#endif
 
+template <typename T>
+struct inspector<T*> {
+    using type = T*;
+    using value_type = T;
+    using base_type = typename inspector<value_type>::base_type;
+    using hdf5_type = typename inspector<value_type>::hdf5_type;
+
+    static constexpr size_t ndim = 1;
+    static constexpr size_t recursive_ndim = ndim + inspector<value_type>::recursive_ndim;
+    static constexpr bool is_trivially_copyable = true;
```

Comment:
From the documentation I gather that this answers the question whether the inner type is trivially copyable? Don't you need the same thing as for `std::vector<T>`?

Reply:
We don't support recursivity with T*, I can try to add it